src/librbd/exclusive_lock/ReleaseRequest.cc

```diff
@@ -130,22 +131,53 @@ Context *ReleaseRequest<I>::handle_flush_notifies(int *ret_val) {
   ldout(cct, 10) << __func__ << dendl;
 
   assert(*ret_val == 0);
-  send_close_journal();
+
+  send_append_journal_event();
   return nullptr;
 }
 
 template <typename I>
-void ReleaseRequest<I>::send_close_journal() {
-  {
-    RWLock::WLocker snap_locker(m_image_ctx.snap_lock);
-    std::swap(m_journal, m_image_ctx.journal);
-  }
+void ReleaseRequest<I>::send_append_journal_event(){
+    CephContext *cct = m_image_ctx.cct;
+    ldout(cct, 10) << __func__ << dendl;
+    {
+      RWLock::WLocker snap_locker(m_image_ctx.snap_lock);
+      std::swap(m_journal, m_image_ctx.journal);
+    }
 
-  if (m_journal == nullptr) {
-    send_close_object_map();
-    return;
-  }
+    if (m_journal == nullptr) {
+      send_close_object_map();
+      return;
+    }
+
+    using klass = ReleaseRequest<I>;
+    Context *ctx = create_context_callback<
+      klass, &klass::handle_append_journal_event>(this);
+    //append journal event
+    uint64_t journal_op_tid = m_image_ctx.journal->allocate_op_tid();
```

Comment:
This is where the unit tests are currently crashing since ```m_image_ctx.journal``` is a nullptr due to the swap on line 145.  Regardless, this will no longer be needed once the new Journal method is added (see below)